.gitlab/maintenance_jobs/kitchen.yml

```diff
@@ -42,3 +42,22 @@ periodic_kitchen_cleanup_azure:
     - RESOURCE_GROUP_PREFIX=kitchen-system-probe python3.6 /deploy_scripts/cleanup_azure.py
     - RESOURCE_GROUP_PREFIX=kitchen-process-collection python3.6 /deploy_scripts/cleanup_azure.py
     - RESOURCE_GROUP_PREFIX=kitchen-container-collection python3.6 /deploy_scripts/cleanup_azure.py
+
+periodic_kitchen_cleanup_ec2:
+  stage: maintenance_jobs
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
+  tags: ["runner:main"]
+  rules:
+    !reference [.manual]
+  script:
+    - export FOUR_HOURS_AGO=`date --date="-4 hours" "+%Y-%m-%d %H:%M:%S"`
+    - export ZOMBIES=`aws ec2 describe-instances --filters "Name=tag:CreatedBy,Values=datadog-agent-kitchen-tests" --region us-east-1 --query "Reservations[*].Instances[?LaunchTime<='$FOUR_HOURS_AGO'].[InstanceId]" --output text`
+    - echo $ZOMBIES
+    # Only run aws ec2 terminate-instances if $ZOMBIES is not empty, otherwise the command fails
+    # https://docs.aws.amazon.com/cli/latest/reference/ec2/terminate-instances.html#options
+    # Note: this fails if $ZOMBIES contains more than 1000 instances (hard limit on the terminate-instances call).
+    # This should rarely (if ever) happens. If it does, manually clean up instances from the AWS console.
+    - |
+      if [ -n "$ZOMBIES_ARRAY" ]; then
```

Comment:
I think this should be `$ZOMBIES` not `$ZOMBIES_ARRAY`

Reply:
You might consider piping the above command into `xargs` and using the `-n` option to break it up into batches. The AWS cli can only handle up to 1000 instance ids, but you may want smaller batches because of this: 
> If you specify multiple instances and the request fails (for example, because of a single incorrect instance ID), none of the instances are terminated.